packages/core/src/classes.ts

```diff
@@ -41,6 +41,11 @@ export type Markup = string
  */
 export type Hyperlink = string
 
+/**
+ * @public
+ */
+export type CollectionSize<T> = T[]['length']
```

Comment:
Slightly more expressive way to declare collection fields. Transpiles to good old `number`, but is more self-explanatory when reading code

Reply:
nice! I would do the same for "sequence" numbers, counters that are used for safe sequential numbers generation